aptiche/lib/main.dart

```diff
@@ -1,23 +1,36 @@
 import 'package:aptiche/utils/theme.dart';
 import 'package:aptiche/views/login/loginscreen.dart';
+import 'package:firebase_core/firebase_core.dart';
 import 'package:flutter/material.dart';
 import 'package:get/get.dart';
 
 void main() {
+  WidgetsFlutterBinding.ensureInitialized();
   runApp(MyApp());
 }
 
 class MyApp extends StatelessWidget {
+  final Future<FirebaseApp> _initialization = Firebase.initializeApp();
   @override
   Widget build(BuildContext context) {
-    return GetMaterialApp(
-      title: "APTI-CHE",
-      debugShowCheckedModeBanner: false,
-      theme: appTheme(),
-      enableLog: true,
-      defaultTransition: Transition.rightToLeftWithFade,
-      popGesture: Get.isPopGestureEnable,
-      home: LoginView(),
-    );
+    return FutureBuilder(
+        future: _initialization,
+        builder: (context, snapshot) {
+          if (snapshot.hasError) {
+            return CircularProgressIndicator();
+          }
+          if (snapshot.connectionState == ConnectionState.done) {
+            return GetMaterialApp(
+              title: "APTI-CHE",
+              debugShowCheckedModeBanner: false,
+              theme: appTheme(),
+              enableLog: true,
+              defaultTransition: Transition.rightToLeftWithFade,
+              popGesture: Get.isPopGestureEnable,
+              home: LoginView(),
```

Comment:
Use `authservice.handleAuth()` instead of direct `LoginView()`  so that the landing screen is on the basis of Auth State